tests/trainer/optimization/test_optimizers.py

```diff
@@ -99,6 +99,19 @@ def test_reducelronplateau_with_no_monitor_in_lr_scheduler_dict_raises(tmpdir):
         trainer.fit(model)
 
 
+def test_onecyclelr_with_epoch_interval_dict_warns(tmpdir):
+    """Test warning when a OneCycleLR is used and interval is epoch."""
+    model = BoringModel()
+    optimizer = optim.Adam(model.parameters())
+    model.configure_optimizers = lambda: {
+        "optimizer": optimizer,
+        "lr_scheduler": {"scheduler": optim.lr_scheduler.OneCycleLR(optimizer, max_lr=0.01, total_steps=3)},
+    }
+    trainer = Trainer(default_root_dir=tmpdir, fast_dev_run=True)
```

Comment:
I think it would be great to split `init_optimizers` into two parts, one static so that this can be tested without instantiating a `Trainer`.

Something like this:

```python
def init_optimizers(self, model: Optional["pl.LightningModule"]) -> Tuple[List, List, List]:
        pl_module = self.lightning_module or model
        self._lightning_optimizers = None
        optim_conf = self.call_hook("configure_optimizers", pl_module=pl_module)
        if optim_conf is None:
            rank_zero_warn(
                "`LightningModule.configure_optimizers` returned `None`, this fit will run with no optimizer",
                UserWarning,
            )
            optim_conf = _MockOptimizer()
        
        optimizers, lr_schedulers, optimizer_frequencies = self._configure_optimizers(optim_conf)
        lr_schedulers = self._configure_schedulers(lr_schedulers, monitor, not pl_module.automatic_optimization)
        _validate_scheduler_optimizer(optimizers, lr_schedulers)
        return optimizers, lr_schedulers, optimizer_frequencies

@staticmethod
def _configure_optimizers(...)

@staticmethod
def _configure_schedulers(...)
```